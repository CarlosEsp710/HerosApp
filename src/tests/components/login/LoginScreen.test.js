import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";

import { LoginScreen } from "../../../components/login/LoginScreen";
import { types } from "../../../types/types";

const mockNavigate = jest.fn();
const mockDispatch = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("test <LoginScreen />", () => {
  const authContext = {
    user: {
      isLoggedIn: false,
    },
    dispatch: mockDispatch,
  };

  test("should return the component <LoginScreen />", () => {
    const wrapper = mount(
      <AuthContext.Provider value={authContext}>
        <MemoryRouter initialEntries={["/login"]}>
          <LoginScreen />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("should call login, navigate and localStorage", () => {
    const wrapper = mount(
      <AuthContext.Provider value={authContext}>
        <MemoryRouter initialEntries={["/login"]}>
          <LoginScreen />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    const action = {
      type: types.login,
      payload: {
        name: "Carlos Espejel",
        email: "carlosespejel71@gmail.com",
        imageUrl: "https://avatars.githubusercontent.com/u/57500554?s=96&v=4",
      },
    };

    wrapper.find("button").simulate("click");

    expect(mockDispatch).toHaveBeenCalled();
    expect(mockDispatch).toHaveBeenCalledWith(action);
    expect(mockNavigate).toHaveBeenCalled();
    expect(mockNavigate).toHaveBeenCalledWith("/", { replace: true });

    localStorage.setItem("lastPath", "/dc");

    wrapper.find("button").simulate("click");

    expect(mockNavigate).toHaveBeenCalledWith("/dc", { replace: true });
  });
});
