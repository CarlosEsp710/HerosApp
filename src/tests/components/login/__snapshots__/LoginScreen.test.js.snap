// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <LoginScreen /> should return the component <LoginScreen /> 1`] = `
<div
  className="min-h-full flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8"
>
  <div
    className="max-w-md w-full space-y-8"
  >
    <div>
      <img
        alt="Workflow"
        className="mx-auto h-12 w-auto"
        src="https://tailwindui.com/img/logos/workflow-mark-indigo-600.svg"
      />
      <h2
        className="mt-6 text-center text-3xl font-extrabold text-gray-900"
      >
        Sign in to your account
      </h2>
      <p
        className="mt-2 text-center text-sm text-gray-600"
      >
        Or
         
        <span
          className="font-medium text-indigo-600 hover:text-indigo-500"
        >
          start your 14-day free trial
        </span>
      </p>
    </div>
    <div
      className="mt-8 space-y-6"
    >
      <input
        defaultValue="true"
        name="remember"
        type="hidden"
      />
      <div
        className="rounded-md shadow-sm -space-y-px"
      >
        <div>
          <label
            className="sr-only"
            htmlFor="email-address"
          >
            Email address
          </label>
          <input
            autoComplete="email"
            className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
            id="email-address"
            name="email"
            placeholder="Email address"
            required={true}
            type="email"
          />
        </div>
        <div>
          <label
            className="sr-only"
            htmlFor="password"
          >
            Password
          </label>
          <input
            autoComplete="current-password"
            className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
            id="password"
            name="password"
            placeholder="Password"
            required={true}
            type="password"
          />
        </div>
      </div>
      <div
        className="flex items-center justify-between"
      >
        <div
          className="flex items-center"
        >
          <input
            className="h-4 w-4 text-indigo-600 focus:ring-indigo-500 border-gray-300 rounded"
            id="remember-me"
            name="remember-me"
            type="checkbox"
          />
          <label
            className="ml-2 block text-sm text-gray-900"
            htmlFor="remember-me"
          >
            Remember me
          </label>
        </div>
        <div
          className="text-sm"
        >
          <span
            className="font-medium text-indigo-600 hover:text-indigo-500"
          >
            Forgot your password?
          </span>
        </div>
      </div>
      <div>
        <button
          className="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
          onClick={[Function]}
        >
          <span
            className="absolute left-0 inset-y-0 flex items-center pl-3"
          >
            <svg
              aria-hidden="true"
              className="h-5 w-5 text-indigo-500 group-hover:text-indigo-400"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M5 9V7a5 5 0 0110 0v2a2 2 0 012 2v5a2 2 0 01-2 2H5a2 2 0 01-2-2v-5a2 2 0 012-2zm8-2v2H7V7a3 3 0 016 0z"
                fillRule="evenodd"
              />
            </svg>
          </span>
          Sign in
        </button>
      </div>
    </div>
  </div>
</div>
`;
